File: 717.1-bit and 2-bit Characters/717.py
class Solution:
    def isOneBitCharacter(self, bits) -> bool:
        flag = True
        for value in bits[:-1]:
            if value == 1:
                if flag == False:
                    flag = True
                else:
                    flag = False
            else:
                flag = True
        return flag

a = Solution()
print(a.isOneBitCharacter([1, 1, 0]))

File: 717.1-bit and 2-bit Characters/717.go
package main

import "fmt"

func main(){
	fmt.Println(isOneBitCharacter([]int{0,1,1,1,0}))
}

func isOneBitCharacter(bits []int) bool {
	bits_length := len(bits)
	index := 0
	for index < bits_length{
		if index == bits_length-1{
			return true
		}
		if bits[index] == 1{
			index += 2
		}else{
			index++
		}
	}
	return false
}